packages/tooltip/src/tooltip.css

```diff
@@ -10,59 +10,253 @@
     clip-path: polygon(0 -5%, 50% 50%, 100% -5%);
-    width: var(--spectrum-tooltip-tip-inline-size) !important;
-    height: var(--spectrum-tooltip-tip-inline-size) !important;
+    clip-path: polygon(0 calc(0% - var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))), 50% var(--mod-tooltip-tip-height-percentage, var(--spectrum-tooltip-tip-height-percentage)), 100% calc(0% - var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))));
+    position: absolute;
+    left: 50%;
+    transform: translateX(-50%);
 }
 
+/** @todo: this needs more documentation */
 #tip[style] {
     transform: none !important;
 }
 
-:host(:not([placement*="top"])) #tooltip {
-    margin-bottom: 0;
+::slotted([slot="icon"]) {
+    inline-size: var(--mod-tooltip-icon-width, var(--spectrum-tooltip-icon-width));
+    block-size: var(--mod-tooltip-icon-height, var(--spectrum-tooltip-icon-height));
+    flex-shrink: 0;
+    align-self: flex-start;
+    margin-block-start: var(--mod-tooltip-icon-spacing-block-start, var(--spectrum-tooltip-icon-spacing-block-start));
+    margin-inline-start: calc(var(--mod-tooltip-icon-spacing-inline-start, var(--spectrum-tooltip-icon-spacing-inline-start)) - var(--mod-tooltip-spacing-inline, var(--spectrum-tooltip-spacing-inline)));
+    margin-inline-end: var(--mod-tooltip-icon-spacing-inline-end, var(--spectrum-tooltip-icon-spacing-inline-end));
+}
+
+#label {
+    line-height: var(--mod-tooltip-line-height, var(--spectrum-tooltip-line-height));
+    margin-block-start: var(--mod-tooltip-spacing-block-start, var(--spectrum-tooltip-spacing-block-start));
+    margin-block-end: var(--mod-tooltip-spacing-block-end, var(--spectrum-tooltip-spacing-block-end));
 }
 
-:host([placement*="top"]) #tooltip #tip {
+:host([placement*="top"]) #tip {
     inset-block-start: 100%;
 }
 
-:host([placement*="bottom"]) #tooltip #tip {
-    inset-block-end: 100%;
-    clip-path: polygon(50% 50%, 0 105%, 100% 105%);
-    top: auto;
+:host([placement*="bottom"]) #tip {
+    clip-path: polygon(50% calc(100% - var(--mod-tooltip-tip-height-percentage, var(--spectrum-tooltip-tip-height-percentage))), 0 calc(100% + var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))), 100% calc(100% + var(--mod-tooltip-tip-antialiasing-inset, var(--spectrum-tooltip-tip-antialiasing-inset))));
+    inset-block: auto 100%;
 }
 
-:host([placement*="left"]) #tooltip #tip,
-:host([placement*="right"]) #tooltip #tip {
+:host([placement="bottom-end"]) #tip,
+:host([placement="bottom-start"]) #tip,
+:host([placement="top-end"]) #tip,
+:host([placement="top-start"]) #tip {
+    transform: none;
+}
+
+:host([placement="bottom-start"]) #tip,
+:host([placement="top-start"]) #tip {
+    inset-inline: var(--mod-tooltip-pointer-corner-spacing, var(--spectrum-tooltip-pointer-corner-spacing)) auto;
+}
```

Comment:
Looks like `top-start`, `top-end`, `bottom-start`, and `bottom-end` all aren't working as expected on the main branch, here they're fixed so that they actually work now 🎉